hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -8,14 +8,285 @@ title: "OTP Controller Technical Specification"
+- Multiple logical partitions of the underlying OTP IP
+	- Each partition is lockable and integrity checked
+	- Integrity digests are stored alongside each logical bank
+- Periodic / persistent checks of OTP values
+	- Periodic checks of shadowed content vs digests
+	- Periodic checks of OTP stored content and shadowed content
+	- Persistent checks for immediate errors
+- Separate life cycle partition and interface to life cycle controller
+	- Supports life cycle functions, but cannot be integrity locked
+- Lightweight scrambling of secret OTP partition using a global netlist constant
+- Lightweight ephemeral key derivation function for RAM scrambling mechanisms
+
+## OTP Controller Overview
+
+The functionality of OTP is split into an open-source and a closed-source part, with a clearly defined boundary in between, as illustrated at a high level in the diagram below.
+
+![OTP Controller Overview](otp_controller_overview.svg)
+
+It is the task of the open-source controller to provide a common, non-technology specific interface to OTP users with a common register interface and a clearly defined I/O interface to hardware.
+The open-source controller implements logical isolation and partitioning of OTP storage that enables users to separate different functions of the OTP into "partitions" with different properties.
+Finally, the open-source controller provides a high level of security for specific partitions by provisioning integrity digests for each partition, and scrambling of partitions where required.
+
+The proprietary IP on the other hand translates a common access interface to the technology-specific OTP interface, both for functional and debug accesses (for example register accesses to the macro-internal control structure).
+
+This split implies that every proprietary OTP IP must implement a translation layer from a standardized OpenTitan interface to the module underneath.
+It also implies that no matter how the OTP storage or word size may change underneath, the open source controller must present a consistent and coherent software and hardware interface.
+This standardized interface is defined further below, and the wrapper leverages the same [technology primitive mechanism]({{< relref "hw/ip/prim/README.md" >}}) that is employed in other parts of OpenTitan in order to wrap and abstract technology-specific macros (such as memories and clocking cells) that are potentially closed-source.
+
+In order to enable simulation and FPGA emulation of the OTP controller even without access to the proprietary OTP IP, a generalized and synthesizable model of the OTP IP is provided in the form of a generic technology primitive (`prim_generic_otp`).
+
 
 # Theory of Operations
 
-## Block Diagram
+Conceptually speaking, the OTP functionality is at a high level split into "front-end" and "back-end".
+The "front-end" contains the logical partitions and associated buffer registers that feed the hardware and software consumer interfaces of the system.
```

Comment:
ah this is probably legacy from the old spec. It implies there that the buffered registers also feed software.  Probably should update that.

Reply:
Yeah, I've actually updated the diagram to show that the "frontend" can have both buffered and unbuffered partitions.
I've removed the `and associated buffer registers` part of this sentence, since further below it is explained that some of the partitions may be buffered, while others may be unbuffered.